app/memoir/routes.py

```diff
@@ -0,0 +1,91 @@
+from flask import Blueprint, request
+from flask_jwt_extended import jwt_required, get_jwt_identity
+from app.memoir import controllers, utils
+from datetime import datetime
+from app.memorial.repos import MemorialRepo
+
+memoir_bp = Blueprint('memoir', __name__, url_prefix="/memoir")
+
+
+# Get all memoirs in a memorial
+@memoir_bp.route('/<memorial_id>/get', methods=["GET"])
+@jwt_required()
+def get_all(memorial_id):
+    memorial_doc = MemorialRepo.get_by_id(memorial_id)
+    if memorial_doc is None:
+        return "Memorial not found", 404
+
+    memoirs = utils.get_all_memoirs(memorial_id)
+
+    return {"memoirs": memoirs}, 200
+
+
+# Get a specific memoir in a memorial
+@memoir_bp.route('/<memorial_id>/<memoir_id>/get', methods=["GET"])
+@jwt_required()
+def get(memorial_id, memoir_id):
+    memorial_doc = MemorialRepo.get_by_id(memorial_id)
+    if memorial_doc is None:
+        return "Memorial not found", 404
+
+    memoir = utils.get_memoir(memorial_id, memoir_id)
+
+    if memoir is None:
+        return "Memoir not found", 404
+
+    return memoir.to_json(), 201
+
+
+@memoir_bp.route('/<memorial_id>/add', methods=["GET"])
```

Comment:
this should be a POST

Reply:
also the `/add` is not necessary either. in general REST api url naming should be nouns & params only since the http protocol already has verbs (GET, POST, PUT, DELETE, etc)